pyannote/database/custom.py

```diff
@@ -271,6 +293,87 @@ def subset_iter(
             {"uri": uri, "database": database, "subset": subset}, lazy=lazy_loader
         )
 
+def subset_trial(
+    self,
+    database: Text = None,
+    task: Text = None,
+    protocol: Text = None,
+    subset: Text = None,
+    entries: Dict = None,
+    database_yml: Path = None,
+):
+    """
+
+    Parameters
+    ----------
+    database : str
+        Database name (e.g. MyDatabase)
+    task : str
+        Task name (e.g. SpeakerDiarization, SpeakerVerification)
+    protocol : str
+        Protocol name (e.g. MyProtocol)
+    subset : {"train_trial", "development_trial", "test_trial"}
+        Subset
+    entries : dict
+        Subset entries.
+    """
+
+    if "trial" not in entries:
+        msg = f"Missing mandatory 'trial' entry in {database}.{task}.{protocol}.{subset}"
+        raise ValueError(msg)
+
+    lazy_loader = dict()
+
+    if 'duration' in entries.keys():
+        def try_with(file: ProtocolFile):
+            return Timeline(segments=[Segment(0, file['duration'])], uri=file['uri'])
+        lazy_loader['try_with'] = try_with
+        
+    elif 'annotated' in entries.keys():
+        def try_with(file: ProtocolFile):
+            return file['annotated']
+        lazy_loader['try_with'] = try_with
+
+    for key, value in entries.items():
+
+        if key == "uri" or key == "trial":
+            continue
+
+        if value.startswith("_"):
+            lazy_loader[key] = Template(value[1:], database_yml)
+
+        else:
+
+            path = resolve_path(Path(value), database_yml)
+
+            # check if file exists
+            if not path.is_file():
+                msg = f"No such file or directory: '{path}'"
+                raise FileNotFoundError(msg)
+
+            # check if loader exists
+            if path.suffix not in LOADERS:
+                msg = f"No loader for file with '{path.suffix}' suffix"
+                raise TypeError(msg)
+
+            # load custom loader class
+            Loader = LOADERS[path.suffix].load()
+            lazy_loader[key] = Loader(path)
+
+    # meant to store and cache once `ProtocolFile` instance per file
+    files: Dict[Text, ProtocolFile] = dict()
+
+    # iterate trials and use preloaded test files
+    for reference, uri1, uri2 in load_trial(resolve_path(Path(entries["trial"]), database_yml)):
```

Comment:
Let's make `load_trial` return just one dict with keys `reference`, `uri1` and `uri2` (to support future _loaders_ that comes with built-in `try_with` keys).

Reply:
I'm note sure I understand, can you elaborate ?

Reply:
I meant updating the signature of `load_trial` so that it returns a single dictionary containing everything:

```python
for trial in load_trial(...):
    reference = trial['reference'] 
    uri1 = trial['uri1']
    uri2 = trial['uri2']
``` 

... just to be compatible with future extensions of `load_trial` that could provide `try_with` directly:
```python
for trial in future_load_trial(...):
    reference = trial['reference'] 
    uri1 = trial['uri1']
    try_with1 = trial['try_with1']
    uri2 = trial['uri2']
    try_with2 = trial['try_with2']
```

Reply:
Done : 6e2cebff8bfcedc6f4510c3f9549799cc86300ca